skimage/feature/blob.py

```diff
@@ -254,29 +275,41 @@ def blob_dog(image, min_sigma=1, max_sigma=50, sigma_ratio=1.6, threshold=2.0,
     gaussian_images = [gaussian_filter(image, s) for s in sigma_list]
 
     # computing difference between two successive Gaussian blurred images
-    # multiplying with standard deviation provides scale invariance
+    # multiplying with average standard deviation provides scale invariance
     dog_images = [(gaussian_images[i] - gaussian_images[i + 1])
-                  * sigma_list[i] for i in range(k)]
+                  * np.mean(sigma_list[i]) for i in range(k)]
 
     image_cube = np.stack(dog_images, axis=-1)
 
     # local_maxima = get_local_maxima(image_cube, threshold)
     local_maxima = peak_local_max(image_cube, threshold_abs=threshold,
                                   footprint=np.ones((3,) * (image.ndim + 1)),
                                   threshold_rel=0.0,
-                                  exclude_border=False)
+                                  exclude_border=exclude_border)
     # Catch no peaks
     if local_maxima.size == 0:
         return np.empty((0, 3))
+
     # Convert local_maxima to float64
     lm = local_maxima.astype(np.float64)
-    # Convert the last index to its corresponding scale value
-    lm[:, -1] = sigma_list[local_maxima[:, -1]]
+
+    # translate final column of lm, which contains the index of the
+    # sigma that produced the maximum intensity value, into the sigma
+    sigmas_of_peaks = sigma_list[local_maxima[:, -1]]
+
+    # if the gaussian is isotropic, the stdev across dimensions are
+    # identical, so return only the stdev deviation of the first dimension
+    if np.unique(min_sigma).shape == (1,) and np.unique(max_sigma).shape == (1,):
+        sigmas_of_peaks = sigmas_of_peaks[:, 0][:, None]
```

Comment:
Can you explain what is happening here? I don't understand the logic here and in the lines above. Also, I think, rather than use `np.unique`, maybe set a flag at the start (when we detect the inputs with `np.isscalar`) and use that here. I can, for example, imagine that someone might want to pass in an array of identical values for sigmas as a way to force the format to be consistent between isotropic and nonisotropic datasets.

Reply:
Oh, right! I momentarily forgot that `np.newaxis` is `None`. Which hints at a slightly preferred solution. ;)

I suggest here, though:

```suggestion
        # select one sigma column, keeping dimension
        sigmas_of_peaks = sigmas_of_peaks[:, 0:1]
```

Reply:
Fixed below!